### ISSUE TITLE
[feat] @eslint-react/naming-convention/use-state Allow destructuring the value only

### ISSUE THREAD

**Issue by bschlenk** [Rel1cx/eslint-react#1342]
### Problem Description

Not sure whether to categorize this as a bug or feature request. There's a pattern of using state without the setter, which sort of acts as a `useRef` that allows an initializer function. It would look like this:

```ts
const [value] = useState(() => expensiveSetup())
```

The `@eslint-react/naming-convention/use-state` flags this value-only state pattern as wrong. To me that seems like a bug. If the rule is meant to enforce a naming convention, but the thing being named is omitted entirely, then there isn't a convention to enforce.

### Alternative Solutions

You can conditionally initialize a ref, but it isn't as succinct, and you then need to do non-null assertions everywhere you read from the ref, or lie and say the ref can't be null.

I can also just `eslint-disable` when I want to use the state with no setter pattern.

```ts
const value = useRef(null)
if (!value.current) {
  value.current = expensiveSetup()
}
```

### Feature Details

I could see the `@eslint-react/naming-convention/use-state` rule either being updated to allow omitting the setter by default, or accepting a config to allow it.

### Examples

```ts
const [value] = useState(() => expensiveSetup())
```

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users

**Comment by silverwind** [comment]
Imho, a `useState` without a setter would better better replaced with a `useMemo` and I would like `useState` that only destructures one value to be continued to be flagged.

The `enforceAssignment` option seems to not work as described, because it does not flag the `const [val] = useState(null)` pattern, which is not a "value and setter pair".

**Comment by silverwind** [comment]
> a useState without a setter would better better replaced with a useMemo

I've ingestigated a bit more, and it's not quite the same because with `useMemo`, react may "throw away the cache" under certain circumstances, which can not happen with `useState`. So I think `useState` without setter has valid use cases that can not be replaced with `useMemo`.
